aya/src/obj/btf/btf.rs

```diff
@@ -308,6 +308,30 @@ impl Btf {
         Ok(s.to_string_lossy())
     }
 
+    pub(crate) fn replace_string_at(&mut self, offset: u32, value: &str) -> Result<(), BtfError> {
+        let btf_header {
+            hdr_len,
+            mut str_off,
+            str_len,
+            ..
+        } = self.header;
+        str_off += hdr_len;
+
+        let c_str = CString::new(value).unwrap();
+        let new = c_str.as_bytes_with_nul();
+
+        if offset >= str_off + str_len || offset + new.len() as u32 > str_off + str_len {
```

Comment:
This needs some tests. The first if condition is a subset of the second so it's redundant.

The API is also a bit odd: it allows replacing a string with a replacement that is the same length or shorter, but not longer. Seems pretty arbitrary.